packages/typespec-ts/src/modular/emitSamples.ts

```diff
@@ -130,6 +132,44 @@ function emitMethodSamples(
     });
   }
 
+  // Check if we need DefaultAzureCredential import
+  const needsCredentialImport =
+    isAzurePackage({ options: dpgContext.rlcOptions }) &&
+    (hasKeyCredential(options.topLevelClient.clientInitialization) ||
+      hasTokenCredential(options.topLevelClient.clientInitialization));
+
+  if (needsCredentialImport) {
+    sourceFile.addImportDeclaration({
+      moduleSpecifier: "@azure/identity",
+      namedImports: ["DefaultAzureCredential"]
+    });
+  }
+
+  // Check if we need dotenv (only if there are server parameters)
+  const hasServerParameters = (() => {
```

Comment:
We should not calculate the client-level parameters by ourselves and pls refer the client constructor logic.